scripts/globals/spells/bluemagic/pinecone_bomb.lua

```diff
@@ -0,0 +1,65 @@
+-----------------------------------------
+-- Spell: Pinecone Bomb
+-- Additional effect: Sleep. Duration of effect varies with TP
+-- Spell cost: 48 MP
+-- Monster Type: Plantoids
+-- Spell Type: Physical (Piercing)
+-- Blue Magic Points: 2
+-- Stat Bonus: STR+1
+-- Level: 36
+-- Casting Time: 2.5 seconds
+-- Recast Time: 26.5 seconds
+-- Skillchain Element(s): Fire (can open Scission or Fusion and can close Liquefaction)
+-- Combos: None
+-----------------------------------------
+require("scripts/globals/bluemagic")
+require("scripts/globals/status")
+require("scripts/globals/magic")
+-----------------------------------------
+
+function onMagicCastingCheck(caster,target,spell)
+    return 0
+end
+
+function onSpellCast(caster,target,spell)
+    local params = {}
+    -- This data should match information on http://wiki.ffxiclopedia.org/wiki/Calculating_Blue_Magic_Damage
+    params.tpmod = TPMOD_CRITICAL
```

Comment:
I was going to say that the TP mod for this spell doesn't seem to be critical hit chance, but then I looked at [the blue magic global](https://github.com/project-topaz/topaz/blob/3df25d52158be6d44dfe84f198194b6822b5ebb8/scripts/globals/bluemagic.lua) and it doesn't look like tpmod params are used _at all_.

I did some searches and [found a definition for `params.tpmod = TPMOD_DURATION` for Battle Dance](https://github.com/project-topaz/topaz/blob/3df25d52158be6d44dfe84f198194b6822b5ebb8/scripts/globals/spells/bluemagic/battle_dance.lua), which seems like what we should be using here (again, this param is never checked), but [that specific global value doesn't seem to be defined anyway](https://github.com/project-topaz/topaz/blob/master/scripts/globals/bluemagic.lua#L4-L9).

I'm not going to expect you to fix that fact that these mods don't seem to do anything, but could you change this to at least _claim_ it's supposed to effect duration?

Reply:
A lot of these BLU enums seems to be in bad shape. I'm happy to clean them up as needed, but need some guidance.

Would you prefer which/any of the following:

- Remove the TPMOD param since it's unused
- Add TPMOD_DURATION value since it's missing, and then reference it here
- Add usage of `params.tpmod` in the global bluemagic.lua (if so, where)

I think the bottom two are probably the *right* approach, assuming this value is supposed to be used somewhere. But if we felt this value is obsolete or unnecessary, perhaps removing it until it's implemented would be less ambiguous?

Also we should probably file an issue for this problem to track these changes, which can be applied ahead of this PR as needed.

Reply:
Filed https://github.com/project-topaz/topaz/issues/354 to track.

Let me know what your preferences are. I'm happy to do an audit or cleanup based on what is preferred. (Although if we want to use the parameters, I could use some pointers since I'm not that familiar with the code and have no retail FFXI)

Reply:
Yeah, the bottom two are the more time consuming, but overall the best choice. It will require a bit of rework for the Blue Magic global, which desperately needs it.

For the purpose of _this_ PR though, go ahead and add TPMOD_DURATION to both the global and this script (matching the one for Battle Dance), and then after a separate PR hooking up the params to be used, we can work on having spells being able to make use of the mods.

Reply:
Sounds like a plan. Will update this in a commit.

Reply:
Commit added. Please take a look.